src/event.js

```diff
@@ -428,75 +398,40 @@ jQuery.event = {
 		return handlerQueue;
 	},
 
-	// Includes all common event props including KeyEvent and MouseEvent specific props
-	props: ( "altKey bubbles cancelable ctrlKey detail eventPhase " +
-		"metaKey shiftKey view which char charCode key keyCode " +
-		"button buttons clientX clientY offsetX offsetY pageX pageY screenX screenY toElement"
-	).split( " " ),
-
-	propHooks: {
-		which: function( event ) {
-			var button = event.button;
-
-			// Add which for key events
-			if ( event.which == null && rkeyEvent.test( event.type ) ) {
-				return event.charCode != null ? event.charCode : event.keyCode;
-			}
-
-			// Add which for click: 1 === left; 2 === middle; 3 === right
-			if ( !event.which && button !== undefined && rmouseEvent.test( event.type ) ) {
-				return ( button & 1 ? 1 : ( button & 2 ? 3 : ( button & 4 ? 2 : 0 ) ) );
-			}
-
-			return event.which;
-		},
-
-		pageX: function( event ) {
-			var eventDoc, doc, body;
-
-			// Calculate pageX if missing and clientX available
-			if ( event.pageX == null && event.clientX != null ) {
-				eventDoc = event.target.ownerDocument || document;
-				doc = eventDoc.documentElement;
-				body = eventDoc.body;
-
-				return event.clientX +
-					( doc && doc.scrollLeft || body && body.scrollLeft || 0 ) -
-					( doc && doc.clientLeft || body && body.clientLeft || 0 );
-			}
-
-			return event.pageX;
-		},
+	addProp: function( name, hook ) {
```

Comment:
> The docs you referenced mention to take care not to clobber anything and give an example

That's true, but the proposed API no longer varies by event type. What kind of clobbering are you concerned about?

> the proposed API doesn't provide any way to ensure that or even know if there was a previous hook.

`Object.prototype.hasOwnProperty.call( jQuery.Event.prototype, name )`

> Maybe a `force` boolean argument could specify whether to clobber an existing hook? I don't think it needs to be any more complicated than that. The scenario I could see is that someone adds a hook that we later incorporate into jQuery with additional functionality, it would work better if their hook was ignored for that case.

Seems like overkill to me, but easy enough to include for I'm guessing about 12 bytes. Does this mean you're also convinced on `jQuery.event.addProp`, though?